src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+  const dateParts = date.split(fromSeparator);
+
+  const mapping = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    const key = fromFormat[i];
+
+    mapping[key] = dateParts[i];
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    mapping['YY'] = mapping['YYYY'].slice(-2);
+  } else if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(mapping['YY'], 10);
+
+    mapping['YYYY'] = year < 30 ? `20${mapping['YY']}` : `19${mapping['YY']}`;
+  }
```

Comment:
The logic for converting between 'YYYY' and 'YY' formats is implemented, but it assumes that the year conversion is only needed when both formats are present. Consider adding error handling or validation to ensure that the input date and formats are consistent and valid.